packages/sdk/src/fetcher/index.ts

```diff
@@ -15,10 +15,10 @@ import { SubgraphFetcher } from "./subgraph";
 import { CoreFetcher } from "./core";
 import { Token } from "../entities/token";
 
+export * from "./types";
 export * from "./abstraction";
-export * from "./core";
 
-class FullFetcher implements IFullCarrotFetcher {
+class FullFetcher extends CoreFetcher implements IFullCarrotFetcher {
```

Comment:
I don't know if makes sense to extends the class just to get a function used one time.

I think I prefer `CoreFetcher.fetchERC20Tokens({ provider, addresses })`

Also why the class is called _CoreFetcher_, why it's _core_?

Reply:
It makes sense because by doing this we don't have to export the `CoreFetcher` + the overall `Fetcher`.

We can only export the general fetcher that will inherit all of the `CoreFetcher` functionality. It's called `CoreFetcher` because it contains logic that is core to Carrot and decently generic, such as resolving templates, fetching data from IPFS, fetching ERC20 tokens etc etc.

Reply:
Yes, noticed how it's used and using only fetcher to access this classes is a good thing.